src/main/java/com/exadel/discount/controller/CouponController.java

```diff
@@ -0,0 +1,74 @@
+package com.exadel.discount.controller;
+
+import com.exadel.discount.dto.coupon.CouponDto;
+import com.exadel.discount.service.CouponService;
+import io.swagger.annotations.ApiOperation;
+import lombok.RequiredArgsConstructor;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.UUID;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/coupons")
+public class CouponController {
+    private final CouponService couponService;
+
+    @GetMapping
+    @ApiOperation("Get list of all coupons")
+    public List<CouponDto> getAllCoupons() {
+        return couponService.findAllCoupons();
+    }
+
+    @GetMapping("{id}")
+    @ApiOperation("Get coupon by ID")
+    public CouponDto getCouponById(@PathVariable final UUID id) {
+        return couponService.findCouponById(id);
+    }
+
+    @PostMapping
+    @ApiOperation("Save new coupon")
+    public CouponDto addCoupon(@RequestParam("userId") final UUID userId,
+                               @RequestParam("discountId") final UUID discountId) {
+        return couponService.assignCouponToUser(userId, discountId);
+    }
+
+    @GetMapping("/ofuser")
+    @ApiOperation("Get coupons of certain user")
+    public List<CouponDto> getCouponsOfUser(@RequestParam("userId") final UUID userId) {
+        return couponService.getCouponsOfUser(userId);
+    }
+
+    //Time example : 2021-06-15T11:58:11
+    @PutMapping
+    @ApiOperation("Edit coupon date")
+    public CouponDto editCouponDate(@RequestParam("couponId") final UUID id,
```

Comment:
Am I right, that when User press button "get discount" it is adding to our database and 'date' field is the date of receiving the coupon? In this case we don't have need of editing date. When user will press this button again it will be a new record in DB, because we need this just for statistics

Reply:
I've written this method for future Archive feature, when over time coupons will be removed to another table archived_coupons.

Reply:
Do U think:remove out ?

Reply:
Done